src/store/slices/switchTheme.ts

```diff
@@ -0,0 +1,20 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { SwitchThemeStateType } from 'src/types';
+
+const initialState: SwitchThemeStateType = {
+  theme: localStorage.getItem('theme') || window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light',
+};
```

Comment:
Nit: in case if JS is off LocalStorage is undefined and there is can be an Error. Don't fix it. Just think over how would you do it.